src/main/native/jni_fast_zstd.c

```diff
@@ -538,16 +532,16 @@ JNIEXPORT jlong JNICALL Java_com_github_luben_zstd_ZstdDecompressCtx_loadDDict0
 /*
  * Class:     com_github_luben_zstd_ZstdDecompressCtx
  * Method:    reset0
- * Signature: (L)J
+ * Signature: (J)J
  */
 JNIEXPORT jlong JNICALL Java_com_github_luben_zstd_ZstdDecompressCtx_reset0
-  (JNIEnv *env, jclass jctx) {
-    ZSTD_DCtx* dctx = (ZSTD_DCtx*)(intptr_t)(*env)->GetLongField(env, jctx, compress_ctx_nativePtr);
```

Comment:
The actual bug was on this line here... but I always find it clearer and less error-prone to pass the fields in a static JNI call rather than look the values up by GetLongField etc... so I refactored the other calls too.